### PR TITLE
fix #16506 by changing the example

### LINKED ISSUE #16506: Asynchttpserver example doesn't work
https://github.com/nim-lang/Nim/blob/95f599ca2d94e4d58055390c4b9c7761bbd60d01/lib/pure/asynchttpserver.nim#L26-L43

Runs out of memory with default gc as well as orc. OS arch linux

```
$ nim -v
Nim Compiler Version 1.5.1 [Linux: amd64]
Compiled at 2020-12-29
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 95f599ca2d94e4d58055390c4b9c7761bbd60d01
active boot switches: -d:release --gc:markAndSweep
```

### PR DESCRIPTION
**Pull request by narimiran**
@dom96 I have no experience with `async`: can you please see if this is a correct fix or this is merely a patch that hides the problem?
If latter, can you please investigate what's going on with this example?

### CODE REVIEW

**Comment by timotheecour** [comment]
this fixes the busyloop (refs https://github.com/nim-lang/Nim/issues/16506#issuecomment-752238571) but `runForever()` is still never reached

File: lib/pure/asynchttpserver.nim [hunk 1]
```diff
@@ -35,7 +35,7 @@
 ##     server.listen Port(8080)
 ##     while true:
 ##       if server.shouldAcceptRequest():
-##         asyncCheck server.acceptRequest(cb)
+##         waitFor server.acceptRequest(cb)
 ##       else:
 ##         poll()
```
Comment (dom96) [inline lib/pure/asynchttpserver.nim]: I'm not sure why the `poll` is here tbh. Can we just remove it?

Edit: discussed with Araq offline.

```suggestion
##         asyncCheck server.acceptRequest(cb)
```

File: lib/pure/asynchttpserver.nim [hunk 2]
```diff
@@ -35,7 +35,7 @@
 ##     server.listen Port(8080)
 ##     while true:
 ##       if server.shouldAcceptRequest():
-##         asyncCheck server.acceptRequest(cb)
+##         waitFor server.acceptRequest(cb)
```
Comment (Araq) [inline lib/pure/asynchttpserver.nim]:
```suggestion
##         await server.acceptRequest(cb)
```

File: lib/pure/asynchttpserver.nim [hunk 3]
```diff
@@ -382,7 +382,7 @@ when not defined(testing) and isMainModule:
     server.listen Port(5555)
     while true:
       if server.shouldAcceptRequest():
-        asyncCheck server.acceptRequest(cb)
+        waitFor server.acceptRequest(cb)
```
Comment (Araq) [inline lib/pure/asynchttpserver.nim]:
```suggestion
        await server.acceptRequest(cb)
```

**Comment by Araq** [comment]
Note: I discussed the patch with Dom and tested it on my machine.

File: lib/pure/asynchttpserver.nim (hunk 1 above)
Comment (timotheecour) [reply lib/pure/asynchttpserver.nim]: the poll is there to avoid exceeding maxFds, https://github.com/nim-lang/Nim/pull/16586 clarifies this

**Comment by dom96** [comment]
This is still wrong though. We discussed `await sleepAsync()` as well. You shouldn't `poll` inside an {.async.}.

**Comment by timotheecour** [comment]
is there a failing test case? IMO there should be a way to set the max number of concurrent connections (this would also help writing such test cases)

**Pull request merged**